crates/templates/src/store.rs

```diff
@@ -0,0 +1,96 @@
+use std::path::{Path, PathBuf};
+
+use anyhow::{anyhow, Context};
+use sha2::Digest;
+
+use crate::directory::subdirectories;
+
+pub(crate) struct TemplateStore {
+    root: PathBuf,
+}
+
+lazy_static::lazy_static! {
+    static ref UNSAFE_CHARACTERS: regex::Regex = regex::Regex::new("[^-_a-zA-Z0-9]").expect("Invalid identifier regex");
+}
+
+impl TemplateStore {
+    pub(crate) fn new(root: impl AsRef<Path>) -> Self {
+        Self {
+            root: root.as_ref().to_owned(),
+        }
+    }
+
+    pub(crate) fn default() -> anyhow::Result<Self> {
+        let data_dir = dirs::data_local_dir()
+            .or_else(|| dirs::home_dir().map(|p| p.join(".fermyon")))
```

Comment:
I think this should probably be `.spin`?

Reply:
That would result in `~/.spin/spin` for folks who don't have a `data_local_dir` though.

This is probably a broader question on where we like to store stuff and merits some bikeshedding.

I now also notice that if the user does _not_ have a `data_local_dir()`, and then later does something that causes them to have one, then their installed templates will disappear.  Not sure whether this is an issue or what to do about it though.

Reply:
A quick perusal of the code suggests that `data_local_dir` won't return `None` for any major OS.